### PR TITLE
Adds ASPNETCORE_SSL_PORT as a way to get port

### PR DESCRIPTION
**Pull request by jkotalik**
Also changes the name from TLSPort to SSLPort. 
For our HTTPS story.

### CODE REVIEW

File: samples/HttpsPolicySample/Startup.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public void ConfigureServices(IServiceCollection services)
             services.AddHttpsRedirection(options =>
             {
                 options.RedirectStatusCode = StatusCodes.Status301MovedPermanently;
-                options.TlsPort = 5001;
+                options.SSLPort = 5001;
```
Comment (javiercn) [inline samples/HttpsPolicySample/Startup.cs]: The correct term is Tls, please revert this change and avoid using SSL

File: src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs [hunk 2]
```diff
@@ -31,10 +32,25 @@ public static IApplicationBuilder UseHttpsRedirection(this IApplicationBuilder a
 
             var options = app.ApplicationServices.GetRequiredService<IOptions<HttpsRedirectionOptions>>().Value;
 
+            // The tls port set in options will have priority over the one in configuration.
+            var tlsPort = options.SSLPort;
+            if (tlsPort == null)
+            {
+                // Only read configuration if there is no tlsPort
+                var config = app.ApplicationServices.GetRequiredService<IConfiguration>();
+                var configTlsPort = config["SSL_PORT"];
```
Comment (javiercn) [inline src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs]: @tratcher Should this be TLS_PORT or TLSPORT or HTTPSPORT or HTTPS_PORT

File: samples/HttpsPolicySample/Startup.cs (hunk 1 above)
Comment (Tratcher) [reply samples/HttpsPolicySample/Startup.cs]: The same property is called SslPort at least two other places in the stack. Consistency is easier to understand. We don't actually know what protocol the underlying server is using, and most people don't know the difference anyways.

Comment (Tratcher) [reply samples/HttpsPolicySample/Startup.cs]: https://github.com/search?l=C%23&q=org%3Aaspnet+SslPort&type=Code
SslPort: launchsettings.json, MvcOptions, RequireHttpsAttribute, AddRedirectToHttps.

Comment (javiercn) [reply samples/HttpsPolicySample/Startup.cs]: Two wrongs don’t make a right. Should we just call it https port moving forward? That way we use the scheme instead of the protocol.

MvcOptions and RequireHttpsAttribute are not recommended anymore. The port on mvcoptions is for development mostly.

Comment (Tratcher) [reply samples/HttpsPolicySample/Startup.cs]: These are all primarily for development.

It doesn't merit a long debate. I appeal to @Eilon as the namer of arbitrary things.

File: src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs (hunk 2 above)
Comment (jkotalik) [reply src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs]: I'm okay with changing it to HTTPS_PORT, but I agree with @Tratcher for SSL > TLS.

File: samples/HttpsPolicySample/Startup.cs (hunk 1 above)
Comment (Eilon) [reply samples/HttpsPolicySample/Startup.cs]: If we're creating new names, I think they should use the correct terminology, i.e. `HTTPS` and not `SSL`. Even though `TLS` might be correct right now, who knows what the future will hold. The term `HTTPS` should hopefully outlive all of those.

Comment (Eilon) [reply samples/HttpsPolicySample/Startup.cs]: (Unless, that is, something is actually referring to a specific lower-level protocol. But that doesn't appear to be the case here.)

**Comment by jkotalik** [comment]
@Tratcher is there a place where we store environment variable strings in our stack? If not, we should consider adding one 😄

File: src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs [hunk 3]
```diff
@@ -31,10 +32,25 @@ public static IApplicationBuilder UseHttpsRedirection(this IApplicationBuilder a
 
             var options = app.ApplicationServices.GetRequiredService<IOptions<HttpsRedirectionOptions>>().Value;
 
+            // The tls port set in options will have priority over the one in configuration.
+            var httpsPort = options.HttpsPort;
+            if (httpsPort == null)
+            {
+                // Only read configuration if there is no tlsPort
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionBuilderExtensions.cs]: httpsPort

**Comment by Tratcher** [comment]
No, there's no central env var constants list. It's too component specific.
